lib/fastlane/plugin/wpmreleasetoolkit/actions/common/buildkite_upload_pipeline_action.rb

```diff
@@ -0,0 +1,74 @@
+module Fastlane
+  module Actions
+    class BuildkiteUploadPipelineAction < Action
+      def self.run(params)
+        env_file = params[:env_file]
+        pipeline_file = params[:pipeline_file]
+        branch = params[:branch]
+        commit = params[:commit]
+
+        UI.user_error!("Pipeline file not found: #{pipeline_file}") unless File.exist?(pipeline_file)
+        UI.user_error!('You should not provide both `branch` and `commit`') if !branch.nil? && commit != 'HEAD'
+        UI.user_error!('This action can only be called from a Buildkite CI build') unless ENV['BUILDKITE'] == 'true'
+
+        UI.message "Uploading pipeline on #{pipeline_file}, #{"branch #{branch}, " if branch}commit #{commit}"
```

Comment:
Technically, we don't "upload a pipeline **on** `release-build.yml`", we "upload a pipeline _named_ `release-build.yml`" 😉 

```suggestion
        UI.message "Uploading pipeline `#{pipeline_file}` with #{"branch `#{branch}`, " if branch}commit `#{commit}`"
```
or
```suggestion
        UI.message "Adding steps from `#{pipeline_file}` to the current build (#{"branch: `#{branch}`, " if branch}commit: `#{commit}`)"
```

Reply:
Not sure why I added the `on` later yesterday 🙃
But I really liked the "adding steps from" suggestion, sounds more accurate to what's going to happen. Updated on bf1916a0b4e953698afa854ff23ce7fe2aa9f8d5 & ad7d098f500197dc869e37df5f2318c553bc1fea (tests).